added function check_host_ssh_auth, get_minion_disk_info and get_started_minions

balamurugana opened the pull request.

Commit:
backend: add check_host_ssh_auth function

This function checks whether the given host is accessible for remote
execution by given credentials.

Signed-off-by: Bala.FA <barumuga@redhat.com>

Commit:
backend: add get_minion_disk_info function

This function fetches disk information from given minions using lsblk
command

Signed-off-by: Bala.FA <barumuga@redhat.com>

Commit:
backend: add get_started_minions function

This function returns minions those are in up state.

* If list of minion is passed optionally, it returns only minions in
  up state in the given minion list.

* If timeout is given (defaults to 60 seconds), it listens to salt
  minion start events for given time and collects started minions.
  Passing 0 to timeout disables this.

Signed-off-by: Bala.FA <barumuga@redhat.com>

ChristinaMeno commented on usm_wrappers/salt_wrapper.py:
```diff
@@ -63,6 +73,53 @@ def accept_minion(minion_id):
+        else:
+            started_minions.add(minion)
+
+    # we might have missed out some start events
```
Doing this twice seems wrong and a great place for abstraction if you do intend to do it twice.

balamurugana replied:
In my testing I have seen this ie this code miss out some events and its required to double check for started minions.

ChristinaMeno commented on usm_wrappers/salt_wrapper.py:
```diff
@@ -63,6 +73,53 @@ def accept_minion(minion_id):
 
+def get_started_minions(minions=[], timeout=60):
+    minion_set = set(minions)
+    up_minions = set(runner.cmd('manage.up', []))
```
timeout may not be honored if this takes a long time. This would be surprising to me.

balamurugana replied:
This is run in salt-master only.  I thought it won't take long time.  I agree this has to be fixed properly.  I will update this in next pull request.

ChristinaMeno commented on usm_wrappers/salt_wrapper.py:
```diff
@@ -63,6 +73,53 @@ def accept_minion(minion_id):
+            break
+        start_time = time.time()
+
+        ret = sevent.get_event(wait=wait, full=True, tag='salt/minion')
```
seems strange to wait for wait time. How will you ever get more than one or two iterations?
I expect this to be a constant period that is some fraction of timeout.

balamurugana replied:
The way get_event works is it waits for an event tag to maximum 'wait' time.  It breaks whenever an event available else it times out.  There is a possibility to get an event quickly than max 'wait' time.

I tried to start with constant period for 'wait' time which doesn't give any improvement.

ChristinaMeno commented on usm_wrappers/salt_wrapper.py:
```diff
@@ -138,6 +195,48 @@ def _check_ip_network(minion, ip, ip_list, network, subnets,
+
+        dev_info = {}
+        for d in devlist:
+            INUSE = True
```
Is there some reason to capitalize INUSE?

balamurugana replied:
My bad.  'in_use' is better than 'INUSE'

ChristinaMeno commented on usm_wrappers/salt_wrapper.py:
```diff
@@ -138,6 +195,48 @@ def _check_ip_network(minion, ip, ip_list, network, subnets,
 
 
+def get_minion_disk_info(minions):
+    columes = 'NAME,KNAME,FSTYPE,MOUNTPOINT,UUID,PARTUUID,MODEL,SIZE,TYPE,' \
```
A doc string would go a long way toward describing what this function expects to return.

balamurugana replied:
Yes.  I will add it in next update

ChristinaMeno commented on usm_wrappers/utils.py:
```diff
@@ -511,3 +511,15 @@ def rexecCmd(cmd, host, port=22, fingerprint=None, username=None,
+        if not rc:
+            return True
+    except SSHCmdExecFailed:
+        pass
```
It is a mistake to not log this failure.

balamurugana replied:
Yes.  It's quite wrong.  I will fix it.

Commit:
backend: add gluster volume functionalities

Signed-off-by: Bala.FA <barumuga@redhat.com>

Commit:
backend: add create_gluster_brick function

Signed-off-by: Bala.FA <barumuga@redhat.com>

Commit:
backend: add function to create ceph pool

Signed-off-by: Bala.FA <barumuga@redhat.com>

Commit:
backend: add function to add bricks to existing volume

Signed-off-by: Bala.FA <barumuga@redhat.com>

Commit:
backend: add function to get gluster volume usage

Signed-off-by: Bala.FA <barumuga@redhat.com>

Commit:
backend: change default pg_num/pgp_num to 128

As per ceph documentation default pg and pgp number is set to 128 for
less than 5 OSDs setup

Signed-off-by: Bala.FA <barumuga@redhat.com>

Commit:
backend: add function to list ceph pools

Signed-off-by: Bala.FA <barumuga@redhat.com>

Commit:
backend: fix osd activation in add_ceph_osd function

Signed-off-by: Bala.FA <barumuga@redhat.com>
